app/controllers/purchases_controller.rb

```diff
@@ -14,9 +14,9 @@ def edit; end
   def create
     @categories = Category.all
     @purchase = Purchase.new(purchase_params)
-
     @purchase.subcategory = Subcategory.find(params[:subcategory_id]) if params[:subcategory_id].present?
-    @purchase.user = User.find(params[:user_id]) if params[:user_id].present?
+    @purchase.purchaser = User.find(params[:purchase][:purchaser_id]) if params[:purchase][:purchaser_id].present?
+    @purchase.purchasee = User.find(params[:purchase][:purchasee_id]) if params[:purchase][:purchasee_id].present?
```

Comment:
I'm just wondering about the `.present?` calls here - I don't know it offhand, but it seems like there should be some way to get around this.